File: simnibs/examples/tDCS_ring.m
%
% example script that runs a simulation
% for a center-surround ring montage
% 
% G. Saturnino, A. Thielscher, 2018
%

%% General information

S = sim_struct('SESSION'); % Define a stimulation sessions
S.fnamehead = 'ernie.msh'; % Choose the head mesh
S.pathfem = 'tdcs_ring'; % Folder for the simulation output

%% Define the tDCS simulation
S.poslist{1} = sim_struct('TDCSLIST');
S.poslist{1}.currents = [1e-3, -1e-3]; % Current going through each channel, in Ampere

%First Electrode (circular, central)
S.poslist{1}.electrode(1).channelnr = 1; % Connect it to the fisrt channel (1 mA)
S.poslist{1}.electrode(1).centre = 'C3'; % Place it over C3
S.poslist{1}.electrode(1).shape = 'ellipse'; % Define it to be elliptical/circular
S.poslist{1}.electrode(1).dimensions = [34, 34]; % Electrode diameter (in mm)
S.poslist{1}.electrode(1).thickness = 4; % Electrode 

%Second Electrode (ring, surrounding)
S.poslist{1}.electrode(2).channelnr = 2; %Connect it to the second channel (-1mA)
S.poslist{1}.electrode(2).centre = 'C3'; % Place it over C3
S.poslist{1}.electrode(2).shape = 'ellipse'; % Elliptical shape
S.poslist{1}.electrode(2).dimensions = [100, 100]; %Diameter of 100mm
S.poslist{1}.electrode(2).thickness = 4;

% Hole
S.poslist{1}.electrode(2).holes = sim_struct('ELECTRODE'); % Define the hole
S.poslist{1}.electrode(2).holes.centre = 'C3'; % Hole is also centered in C3
S.poslist{1}.electrode(2).holes.shape = 'ellipse'; % Shape of the hole
S.poslist{1}.electrode(2).holes.dimensions = [75, 75]; % Diameter of 75mm

%% Run Simulation
run_simnibs(S);

%% Visualize Simulations
m = mesh_load_gmsh4(fullfile(S.pathfem, 'ernie_TDCS_1_scalar.msh'));
mesh_show_surface(m);

File: simnibs/examples/simulations/tDCS_ring.py
''' Example of how to run a SimNIBS tDCS simulation with ring electrodes in Python
    Run with:

    simnibs_python example_tDCS_ernie_ring.py

    Copyright (C) 2018 Guilherme B Saturnino
'''
from simnibs import sim_struct, run_simnibs

### General Infoarmation
S = sim_struct.SESSION()
S.subpath = 'm2m_ernie'  # subject folder
S.pathfem = 'tdcs_ring'  # Directory for the simulation


### Define tDCS simulation
tdcs = S.add_tdcslist()
tdcs.currents = [0.001, -0.001]  # Current flow though each channel (mA)

# First Electrode (center, circular)
central_electrode = tdcs.add_electrode()
central_electrode.channelnr = 1  # Connect the electrode to the first channel
central_electrode.centre = 'C3'  # Place it over C3
central_electrode.shape = 'ellipse'  # Elliptical / circular shape
central_electrode.dimensions = [34, 34]  # 34 mm diameter
central_electrode.thickness = 4  # 4 mm thickness

# Second electrode (surround, ring)
ring_electrode = tdcs.add_electrode()
ring_electrode.channelnr = 2
ring_electrode.centre = 'C3'  # Place it over C3
ring_electrode.shape = 'ellipse'  # Elliptical / circular shape
ring_electrode.dimensions = [100, 100]  # Ring outer diameter
ring_electrode.thickness = 4  # 4 mm thickness

hole = ring_electrode.add_hole()
hole.centre = 'C3'
hole.shape = 'ellipse'
hole.dimensions = [75, 75]  # Ring inner diameter
hole.thickness = 4

run_simnibs(S)
